xarray/core/computation.py

```diff
@@ -408,7 +434,20 @@ def apply_dict_of_variables_vfunc(
 
     result_vars = {}
     for name, variable_args in zip(names, grouped_by_name):
-        result_vars[name] = func(*variable_args)
+        core_dim_check = _check_core_dims(signature, variable_args, name)
+        if core_dim_check is True:
```

Comment:
nit: 

```suggestion
        core_dim_present = _check_core_dims(signature, variable_args, name)
        if core_dim_present is True:
```

but feel free to ignore if you don't like it.

Reply:
Better, thanks!